cpp/csp/adapters/websocket/ClientInputAdapter.h

```diff
@@ -16,17 +16,20 @@ class ClientInputAdapter final: public PushInputAdapter {
         Engine * engine,
         CspTypePtr & type,
         PushMode pushMode,
-        const Dictionary & properties
+        const Dictionary & properties,
+        bool dynamic
     );
 
     void processMessage( void* c, size_t t, PushBatch* batch );
+    void processMessage( std::tuple<std::string, void*> data, size_t t, PushBatch* batch );
```

Comment:
OOC: why allocate a tuple here rather than make the call site `processMessage( const std::string& connectionUri, void* c, ...)` ?

Reply:
I think you're right, I totally should just do that